src/database.ts

```diff
@@ -3241,41 +3275,42 @@ class Database extends common.GrpcServiceObject {
           transaction!.excludeTxnFromChangeStreams();
         }
 
-        const release = () => {
+        // Our span should only be ended if the
+        // transaction either errored or was ended.
+        transaction!.once('error', err => {
```

Comment:
I think this might cause some issue, as we do a retry of whole transaction in case of certain errors like "abort" ,

Reply:
See my below comment, if we can handle span.end() using finally block ?

Reply:
Alright, done and updated!

Reply:
We cannot use the finally block because that causes incorrect behavior in which the database.runTransaction span shall only be ended on exiting the callback instead of actually when transaction.end() or transaction.commit() are invoked, if a customer adds a sleep in their code, after the transaction ends, the span won't be correctly ended until the exit of the function. Are you suggesting instead  that we remove `transaction.once('error')`? We need a bare minimum of transaction.once('end') at least.